compiler/test/stdlib/number.test.gr

```diff
@@ -134,3 +134,9 @@ assert Result.isErr(Number.parseInt("zzzzz", 37))
 assert Result.isErr(Number.parseInt("zzzzz", 9223372036854775807))
 assert Result.isErr(Number.parseInt("10", 1.23))
 assert Result.isErr(Number.parseInt("10", 2/3))
+
+// Number.sign
+assert Number.sign(-10000) == -1
+assert Number.sign(22222) == 1
+assert Number.sign(0) == 0
+assert Number.sign(-0) == 0
```

Comment:
Implementation is great, though these aren't quite checking the behavior because of int/float weirdness :/

Signedness only applies to floats, so we'll need to use `0.0` and `-0.0`. Additionally, positive/negative zero in floating point arithmetic are equal, so we'll need to check equality some other way. The best I could think of was dividing 1 by the result of `Number.sign` and checking for positive/negative infinity.

Reply:
We chatted about this in discord and I'm going to implement the tests we came up with, give me a bit.

Reply:
Tests updated. I also added TODO comments to switch to literals once #693 is done!